src/Routes/ListRoutesCommand.php

```diff
@@ -0,0 +1,231 @@
+        "user,user.register,*.register,user*".
+        EOT;
+
+    private const HELP_OPT_HAS_PATH = <<<'EOT'
+        Filter out routes by path. This option accepts a comma-separated list of 
+        one or more paths. The paths can be a fixed-string or a regular expression, 
+        supported by the preg_* functions. For example, "/,/api/ping,*/ping".
+        EOT;
+
+    private const HELP_OPT_SORT = <<<'EOT'
+        Sort the command's output. The supported values are "name" and "path".
+        EOT;
+
+    private const HELP_OPT_SUPPORTS_METHOD = <<<'EOT'
+        Filters out routes by HTTP method.  This option accepts a comma-separated 
+        list of one or more HTTP methods.
+        EOT;
+
+    /** @var null|string Cannot be defined explicitly due to parent class */
+    public static $defaultName = 'mezzio:routes:list';
+
+    public function __construct(RouteCollector $routeCollector)
+    {
+        $this->routes = $routeCollector->getRoutes();
+
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription("Print the application's routing table.");
+        $this->setHelp(self::HELP);
+
+        $this->addOption(
+            'format',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_FORMAT,
+            'table'
+        );
+
+        $this->addOption(
+            'sort',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_SORT,
+            'name'
+        );
+
+        // Routing table filter options
+        $this->addOption(
+            'has-middleware',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_HAS_MIDDLEWARE,
+            false
+        );
+        $this->addOption(
+            'has-name',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_HAS_NAME,
+            false
+        );
+        $this->addOption(
+            'has-path',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_HAS_PATH,
+            false
+        );
+        $this->addOption(
+            'supports-method',
+            null,
+            InputOption::VALUE_REQUIRED,
+            self::HELP_OPT_SUPPORTS_METHOD,
+            false
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $result = 0;
+        if (empty($this->routes)) {
+            $output->writeln("There are no routes in the application's routing table.");
+            return $result;
+        }
+
+        $format = strtolower((string) $input->getOption('format'));
+
+        $sorter = $this->getSortOrder($input) === 'name'
+            ? new RouteSorterByName()
+            : new RouteSorterByPath();
+        usort($this->routes, $sorter);
+
+        $this->filterOptions = [
+            'method'     => strtolower((string) $input->getOption('supports-method')),
```

Comment:
I suggest asserting on the values, rather than casting

Reply:
Yah, I would suggest using https://github.com/php-fig/http-message-util/blob/master/src/RequestMethodInterface.php and create proper assertions or return `ANY` which might be a constant only in this class